### PR TITLE
Update soft delete middleware docs about `findFirstOrThrow` and `findUniqueOrThrow`

### LINKED ISSUE #5391: Update soft delete middleware docs about `findFirstOrThrow` and `findUniqueOrThrow`
While working on my project and referring to the official docs, I used a soft delete middleware and started using `findFirstOrThrow` and `findUniqueOrThrow`. However, I noticed that these two functions do not apply soft delete.

At that time, the prisma/client version I was using missed the `prismaAction` type for `findFirstOrThrow` and `findUniqueOrThrow`, preventing me from applying this feature.

However, through https://github.com/prisma/prisma/pull/17471, 
these two queries were added to the `prismaAction` type in prisma/client version 5.1.0.

Currently, there is no information related to this in the soft delete documentation.
https://www.prisma.io/docs/concepts/components/prisma-client/middleware/soft-delete-middleware#option-2-use-middleware-to-determine-the-behavior-of-readupdate-queries-for-deleted-records

Therefore, we should add information regarding how to enable soft delete middleware with `findFirstOrThrow` and `findUniqueOrThrow` from Prisma/client version 5.1.0 onwards.
```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by Kwontaehwon**
## Describe this PR

While working on my project and referring to the official docs, I used a soft delete middleware and started using `findFirstOrThrow` and `findUniqueOrThrow`. However, I noticed that these two queries can't apply soft delete.

At that time, the prisma/client version I was using missed the `prismaAction` type for `findFirstOrThrow` and `findUniqueOrThrow`, preventing me from applying this feature.

However, through https://github.com/prisma/prisma/pull/17471, 
these two queries were added to the `prismaAction` type in prisma/client version 5.1.0.

Currently, there is no information related to this in the soft delete documentation.
https://www.prisma.io/docs/concepts/components/prisma-client/middleware/soft-delete-middleware#option-2-use-middleware-to-determine-the-behavior-of-readupdate-queries-for-deleted-records

Therefore, we should add information regarding how to enable soft delete middleware with `findFirstOrThrow` and `findUniqueOrThrow` from Prisma/client version 5.1.0 onwards.

## Changes

- Add information about soft delete middleware with `findFirstOrThrow` and `findUniqueOrThrow` is available after 5.1.0 version.

## What issue does this fix?

Fixes #5391

### CODE REVIEW

File: content/200-concepts/100-components/02-prisma-client/054-middleware/100-soft-delete-middleware.mdx [hunk 1]
```diff
@@ -301,6 +301,8 @@ Option two uses Prisma middleware to prevent soft deleted records from being ret
 | `update`     | 🔧 Change query to `updateMany` (because you cannot apply `deleted: false` filters to `update`) <br /> 🔧 Add `where: { deleted: false }` filter to exclude soft deleted posts                                                                                                                                                                                                                                          | `{ count: n }` instead of `Post` |
 | `updateMany` | 🔧 Add `where: { deleted: false }` filter to exclude soft deleted posts                                                                                                                                                                                                                                                                                                                                                 | No change                        |
 
+- **Is it not possible to utilize soft delete with `findFirstOrThrow` or `findUniqueOrThrow`?**<br />
+  From version 5.1.0(https://github.com/prisma/prisma/releases/5.1.0), you can apply soft delete `findFirstOrThrow` or `findUniqueOrThrow` by using middleware.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/054-middleware/100-soft-delete-middleware.mdx]:
```suggestion
  From version [5.1.0](https://github.com/prisma/prisma/releases/5.1.0), you can apply soft delete `findFirstOrThrow` or `findUniqueOrThrow` by using middleware.
```

**Comment by janpio** [comment]
Should we also update the example on the page to cover these then?

**Comment by Kwontaehwon** [comment]
> Should we also update the example on the page to cover these then?

@janpio Thanks for review! I changed example code to support `findFirstOrThrow` and `findUniqueOrThrow`. I also add test about it.

**Comment by jharrell** [comment]
Thank you very much for the contribution @Kwontaehwon ! I'm running our test suite to confirm that everything looks good and then will get this merged.

File: content/200-concepts/100-components/02-prisma-client/054-middleware/100-soft-delete-middleware.mdx [hunk 2]
```diff
@@ -310,6 +312,7 @@ Run the following sample to see how middleware affects each query:
 
 ```ts
 import { PrismaClient } from '@prisma/client'
+import { PrismaClientKnownRequestError } from '@prisma/client/runtime/library'
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/054-middleware/100-soft-delete-middleware.mdx]: I think this is wrong and should import from the package directly.

Comment (Kwontaehwon) [reply content/200-concepts/100-components/02-prisma-client/054-middleware/100-soft-delete-middleware.mdx]: > I think this is wrong and should import from the package directly.

Absolutely agree. I fix code to directly import that error.
Thank you.

Comment (janpio) [reply content/200-concepts/100-components/02-prisma-client/054-middleware/100-soft-delete-middleware.mdx]: Out of interest: How did you end up with this import? Did your IDE suggest this?

Comment (Kwontaehwon) [reply content/200-concepts/100-components/02-prisma-client/054-middleware/100-soft-delete-middleware.mdx]: > Out of interest: How did you end up with this import? Did your IDE suggest this?

yes. I use visual studio code and it suggested this.
I didn't know there was a better way. Thank you for your review. 😄

**Review by jharrell — APPROVED** [review]

**Comment by Kwontaehwon** [comment]
@jharrell It seems like something went wrong during the distribution process. This is my first contribution so I don't know what to do. What should I do?

**Comment by jharrell** [comment]
@Kwontaehwon that's an issue on our side. Everything looks good and I'm merging. Thank you again for your contribution!

**Pull request merged**